rosette/rbl/rosette/space.rbl

```diff
@@ -377,21 +392,23 @@
                 ]
             ]
             (map continuation_group (proc [m [[unification_ptrn binding] continuations_struct]]
-                (map continuations_struct (proc [n [product_ptrns_struct ctxt_triple]]
-                    (let*
-                        [
-                            [channel_position (tuple-safe-nth current_channel_positions i)]
-                                ;;; TODO: This assumes that the channel cannot be in multiple positions in the consume, which is not necessarily true.
-                            [[product_ptrn stored_product] (tuple-safe-nth product_ptrns_struct channel_position)]
-                        ]
-                        (if (match? NameSpace product_ptrn product)
+                (filter-niv
+                    (map continuations_struct (proc [n [product_ptrns_struct ctxt_triple]]
+                        (let*
                             [
-                                [channel (tuple-safe-nth subspaces i)]
-                                [unification_ptrn unification_ptrn]
-                                [product [n channel_position product_ptrn]]
+                                [channel_position (tuple-safe-nth current_channel_positions i)]
+                                    ;;; TODO: This assumes that the channel cannot be in multiple positions in the consume, which is not necessarily true.
```

Comment:
In particular, in the Mercury tutorial, there's the following:
```
new coins in {
    coins!!(1) | coins!!(5) | coins!!(10) |
    for (@(x && ~5) <- coins; @(y && ~10) <- coins if x+y == 15) {
        system!("print", (x, y))
    }
}
```

Reply:
Pretty sure that comment isn't mine.